vcs/gitcmd/repo.go

```diff
@@ -221,6 +235,30 @@ func (r *Repository) Branches(opt vcs.BranchesOptions) ([]*vcs.Branch, error) {
 	return branches, nil
 }
 
+func (r *Repository) branchesWithCommit(commitID string) (map[string]bool, error) {
+	cmd := exec.Command("git", "branch", "--contains", commitID)
+	cmd.Dir = r.Dir
+	out, err := cmd.Output()
+	if err != nil {
+		return nil, err
+	}
+
+	lines := strings.Split(string(out), "\n")
+	branches := make([]string, 0)
+	for _, line := range lines {
+		if len(line) >= 2 {
+			branches = append(branches, line[2:])
+		}
+	}
+
+	branches_ := make(map[string]bool)
```

Comment:
It looks like both the intermediate map and slice can be eliminated, unless you really want to do branch name de-duplication. I don't think that should be necessary.

The entire `refs, err := r.showRef("--heads")` step appears to be unnecessary when you have non-empty `ContainsCommit`. If I'm not mistaken, you can just use the result of `git branch --contains <commit_id>` directly as the source of `[]*vcs.Branch`.